corehq/apps/users/util.py

```diff
@@ -126,18 +126,12 @@ def user_id_to_username(user_id, use_name_if_available=False):
         return raw_username(user_object['username']) if "username" in user_object else None
 
 
+@quickcache(['user_id'])
 def cached_user_id_to_username(user_id):
-    if not user_id:
+    if not user_id or (user_id in WEIRD_USER_IDS and user_id != DEMO_USER_ID):
         return None
```

Comment:
I think this logic belongs in `user_id_to_username`. And I think it should probably return `user_id` (I think this currently changes the API for demo users based on this line: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/users/util.py#L116-L117). Given the intention is likely "display something useful based on something less useful".

Reply:
I think I independently came to the same conclusion that `DEMO_USER_ID` needs to be exempt here and fixed it maybe around the same time you were leaving the comment, because I parts of your comment don't make sense to me based on the code as it looks in the PR now (but would have a commit or two back). The part about moving it into `user_id_to_username` makes total sense though. Just wanted to check with you that I'm understanding the comment right. If so I can make that cleanup change. (There's not a ton of urgency on this PR, the whole thing is just kind of cleanup.)

Reply:
Oops, my bad, you're totally right. I think I started reviewing by commit and then somehow didn't notice the change.